### ISSUE TITLE
osd-prepare does not accept a partition it just prepared

### ISSUE THREAD

**Issue by dkrizic** [rook/rook#5602]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

OSD prepare prepares the disk using LVM, but then it does not take it, so OSD is not provisioned

**Expected behavior:**

OSD prepare should prepare the disk (in my case partition) and start an OSD for it.

**How to reproduce it (minimal and precise):**

The node is kube2. The disks before provisioning:

```
root@kube2:~# uname -a
Linux kube2 4.15.0-101-generic #102-Ubuntu SMP Mon May 11 10:07:26 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
root@kube2:~# lsblk
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda      8:0    0   477G  0 disk 
├─sda1   8:1    0  93.1G  0 part /
└─sda2   8:2    0 383.8G  0 part 
rbd0   252:0    0    10G  0 disk /var/lib/kubelet/pods/cb64b441-8dc3-4ec7-bbf9-e5a69fcb307d/volumes/ceph.rook.io~
rbd1   252:16   0    25G  0 disk /var/lib/kubelet/pods/cb64b441-8dc3-4ec7-bbf9-e5a69fcb307d/volumes/ceph.rook.io~
root@kube2:~# blkid
/dev/sda1: UUID="0548da66-7c3a-4a4e-83a0-287d28ddeb2d" TYPE="xfs" PARTUUID="7a91217b-01"
/dev/sda2: PARTUUID="7a91217b-02"
/dev/rbd0: UUID="eeb5913f-bb67-4682-940c-cb547954f719" TYPE="xfs"
/dev/rbd1: UUID="2593e640-8aa5-46c4-b6f6-0e26c9d8812e" TYPE="xfs"
```

The cluster CRD

```
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  cephVersion:
    image: ceph/ceph:v14.2.9
    allowUnsupported: false
  dataDirHostPath: /var/lib/rook
  mgr:
    modules:
    - name: pg_autoscaler
      enabled: true
  mon:
    count: 4
    allowMultiplePerNode: true
  dashboard:
    enabled: true
    port: 8080
    ssl: false
  network:
    hostNetwork: false
  rbdMirroring:
    workers: 0
  storage:
    useAllNodes: false
    useAllDevices: true
    config:
      databaseSizeMB: "1024" # this value can be removed for environments with normal sized disks (100 GB or larger)
      journalSizeMB: "1024"  # this value can be removed for environments with normal sized disks (20 GB or larger)
      osdsPerDevice: "1" # this value can be overridden at the node or device level
    directories:
    - path: /var/lib/rook
    nodes:
    - name: "kube0"
    - name: "kube1"
      directories:
      - path: "/media/local"
    - name: "kube2"
```

The last line was added. 

Output from the operator after applying the cluster CRD:

```
2020-06-05 08:52:24.426956 I | op-cluster: The Cluster CR has changed. diff=  v1.ClusterSpec{
  	CephVersion: v1.CephVersionSpec{Image: "ceph/ceph:v14.2.9"},
  	Storage: v1.StorageScopeSpec{
  		Nodes: []v1.Node{
  			{Name: "kube0"},
  			{Name: "kube1", Selection: v1.Selection{Directories: []v1.Directory{{Path: "/media/local"}}}},
+ 			{Name: "kube2"},
  		},
  		UseAllNodes: false,
  		NodeCount:   0,
  		... // 4 identical fields
  	},
  	Annotations: nil,
  	Placement:   nil,
  	... // 16 identical fields
  }
2020-06-05 08:52:24.427029 I | op-cluster: update event for cluster "rook-ceph" is supported, orchestrating update now
2020-06-05 08:52:24.450961 I | op-config: CephCluster "rook-ceph" status: "Updating". "Cluster is updating"
2020-06-05 08:52:29.004571 I | op-cluster: ceph daemons running versions are: {Mon:map[ceph version 14.2.9 (581f22da52345dba46ee232b73b990f06029a2a0) nautilus (stable):4] Mgr:map[ceph version 14.2.9 (581f22da52345dba46ee232b73b990f06029a2a0) nautilus (stable):1] Osd:map[ceph version 14.2.9 (581f22da52345dba46ee232b73b990f06029a2a0) nautilus (stable):2] Rgw:map[ceph version 14.2.9 (581f22da52345dba46ee232b73b990f06029a2a0) nautilus (stable):1] Mds:map[ceph version 14.2.9 (581f22da52345dba46ee232b73b990f06029a2a0) nautilus (stable):2] RbdMirror:map[] Overall:map[ceph version 14.2.9 (581f22da52345dba46ee232b73b990f06029a2a0) nautilus (stable):10]}
2020-06-05 08:52:29.040750 I | op-config: CephCluster "rook-ceph" status: "Updating". "Cluster is updating"
2020-06-05 08:52:29.167404 I | op-mon: start running mons
2020-06-05 08:52:29.222420 I | op-mon: parsing mon endpoints: q=10.109.190.149:6789,r=10.107.121.159:6789,m=10.110.175.97:6789,n=10.109.159.46:6789
2020-06-05 08:52:29.297922 I | op-mon: saved mon endpoints to config map map[csi-cluster-config-json:[{"clusterID":"rook-ceph","monitors":["10.109.190.149:6789","10.107.121.159:6789","10.110.175.97:6789","10.109.159.46:6789"]}] data:m=10.110.175.97:6789,n=10.109.159.46:6789,q=10.109.190.149:6789,r=10.107.121.159:6789 mapping:{"node":{"m":{"Name":"kube0","Hostname":"kube0","Address":"10.10.2.20"},"n":{"Name":"kube1","Hostname":"kube1","Address":"10.10.2.21"},"q":{"Name":"kube11","Hostname":"kube11","Address":"10.10.0.98"},"r":{"Name":"kube12","Hostname":"kube12","Address":"10.10.0.97"}}} maxMonId:17]
2020-06-05 08:52:29.368719 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-06-05 08:52:29.369793 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2020-06-05 08:52:29.460820 I | op-mon: targeting the mon count 4
2020-06-05 08:52:51.220134 I | op-mon: checking for basic quorum with existing mons
2020-06-05 08:52:51.446265 I | op-mon: mon "r" endpoint are [v2:10.107.121.159:3300,v1:10.107.121.159:6789]
2020-06-05 08:52:51.654423 I | op-mon: mon "m" endpoint are [v2:10.110.175.97:3300,v1:10.110.175.97:6789]
2020-06-05 08:52:51.963649 I | op-mon: mon "n" endpoint are [v2:10.109.159.46:3300,v1:10.109.159.46:6789]
2020-06-05 08:52:52.211316 I | op-mon: mon "q" endpoint are [v2:10.109.190.149:3300,v1:10.109.190.149:6789]
2020-06-05 08:52:52.447419 I | op-mon: saved mon endpoints to config map map[csi-cluster-config-json:[{"clusterID":"rook-ceph","monitors":["10.109.190.149:6789","10.107.121.159:6789","10.110.175.97:6789","10.109.159.46:6789"]}] data:q=10.109.190.149:6789,r=10.107.121.159:6789,m=10.110.175.97:6789,n=10.109.159.46:6789 mapping:{"node":{"m":{"Name":"kube0","Hostname":"kube0","Address":"10.10.2.20"},"n":{"Name":"kube1","Hostname":"kube1","Address":"10.10.2.21"},"q":{"Name":"kube11","Hostname":"kube11","Address":"10.10.0.98"},"r":{"Name":"kube12","Hostname":"kube12","Address":"10.10.0.97"}}} maxMonId:17]
2020-06-05 08:52:53.250013 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-06-05 08:52:53.250480 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2020-06-05 08:52:53.251111 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-06-05 08:52:53.251850 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2020-06-05 08:52:53.335426 I | op-mon: deployment for mon rook-ceph-mon-r already exists. updating if needed
2020-06-05 08:52:53.396863 I | op-mon: checking if we can stop the deployment rook-ceph-mon-r
2020-06-05 08:53:02.121115 I | op-k8sutil: updating deployment "rook-ceph-mon-r"
2020-06-05 08:53:04.230511 I | op-k8sutil: finished waiting for updated deployment "rook-ceph-mon-r"
2020-06-05 08:53:04.230596 I | op-mon: checking if we can continue the deployment rook-ceph-mon-r
2020-06-05 08:53:04.230661 I | op-mon: waiting for mon quorum with [r m n q]
2020-06-05 08:53:04.399667 I | op-mon: mons running: [r m n q]
2020-06-05 08:53:09.534629 I | op-mon: Monitors in quorum: [m n q r]
2020-06-05 08:53:09.565768 I | op-mon: deployment for mon rook-ceph-mon-m already exists. updating if needed
2020-06-05 08:53:09.687737 I | op-mon: checking if we can stop the deployment rook-ceph-mon-m
2020-06-05 08:53:19.036197 I | op-k8sutil: updating deployment "rook-ceph-mon-m"
2020-06-05 08:53:21.148595 I | op-k8sutil: finished waiting for updated deployment "rook-ceph-mon-m"
2020-06-05 08:53:21.148701 I | op-mon: checking if we can continue the deployment rook-ceph-mon-m
2020-06-05 08:53:21.148789 I | op-mon: waiting for mon quorum with [r m n q]
2020-06-05 08:53:21.391129 I | op-mon: mons running: [r m n q]
2020-06-05 08:53:26.869440 I | op-mon: Monitors in quorum: [m n q r]
2020-06-05 08:53:26.907900 I | op-mon: deployment for mon rook-ceph-mon-n already exists. updating if needed
2020-06-05 08:53:26.968766 I | op-mon: checking if we can stop the deployment rook-ceph-mon-n
2020-06-05 08:53:35.809905 I | op-k8sutil: updating deployment "rook-ceph-mon-n"
2020-06-05 08:53:37.916982 I | op-k8sutil: finished waiting for updated deployment "rook-ceph-mon-n"
2020-06-05 08:53:37.917093 I | op-mon: checking if we can continue the deployment rook-ceph-mon-n
2020-06-05 08:53:37.917174 I | op-mon: waiting for mon quorum with [r m n q]
2020-06-05 08:53:38.069705 I | op-mon: mons running: [r m n q]
2020-06-05 08:53:43.257141 I | op-mon: Monitors in quorum: [m n q r]
2020-06-05 08:53:43.307491 I | op-mon: deployment for mon rook-ceph-mon-q already exists. updating if needed
2020-06-05 08:53:43.363185 I | op-mon: checking if we can stop the deployment rook-ceph-mon-q
2020-06-05 08:53:52.102188 I | op-k8sutil: updating deployment "rook-ceph-mon-q"
2020-06-05 08:53:54.183568 I | op-k8sutil: finished waiting for updated deployment "rook-ceph-mon-q"
2020-06-05 08:53:54.183681 I | op-mon: checking if we can continue the deployment rook-ceph-mon-q
2020-06-05 08:53:54.183744 I | op-mon: waiting for mon quorum with [r m n q]
2020-06-05 08:53:54.353769 I | op-mon: mons running: [r m n q]
2020-06-05 08:53:59.545031 I | op-mon: Monitors in quorum: [m n q r]
2020-06-05 08:53:59.545137 I | op-mon: mons created: 4
2020-06-05 08:54:04.001205 I | op-mon: waiting for mon quorum with [r m n q]
2020-06-05 08:54:04.151240 I | op-mon: mons running: [r m n q]
2020-06-05 08:54:09.111690 I | op-mon: Monitors in quorum: [m n q r]
2020-06-05 08:54:09.111895 I | cephclient: getting or creating ceph auth key "client.csi-rbd-provisioner"
2020-06-05 08:54:14.618905 I | cephclient: getting or creating ceph auth key "client.csi-rbd-node"
2020-06-05 08:54:19.767315 I | cephclient: getting or creating ceph auth key "client.csi-cephfs-provisioner"
2020-06-05 08:54:24.822489 I | cephclient: getting or creating ceph auth key "client.csi-cephfs-node"
2020-06-05 08:54:30.066823 I | ceph-csi: created kubernetes csi secrets for cluster "rook-ceph"
2020-06-05 08:54:30.066996 I | cephclient: getting or creating ceph auth key "client.crash"
2020-06-05 08:54:35.302934 I | ceph-crashcollector-controller: created kubernetes crash collector secret for cluster "rook-ceph"
2020-06-05 08:54:40.199574 I | cephclient: successfully enabled msgr2 protocol
2020-06-05 08:54:40.199776 I | op-mgr: start running mgr
2020-06-05 08:54:40.199864 I | cephclient: getting or creating ceph auth key "mgr.a"
2020-06-05 08:54:45.274699 I | op-mgr: deployment for mgr rook-ceph-mgr-a already exists. updating if needed
2020-06-05 08:54:45.346439 I | op-k8sutil: deployment "rook-ceph-mgr-a" did not change, nothing to update
2020-06-05 08:54:45.643607 I | op-mgr: dashboard service already exists
2020-06-05 08:54:45.903165 I | op-mgr: mgr metrics service already exists
2020-06-05 08:54:45.903994 I | op-osd: start running osds in namespace rook-ceph
2020-06-05 08:54:45.904529 I | op-osd: start provisioning the osds on pvcs, if needed
2020-06-05 08:54:45.904988 I | op-osd: no volume sources defined to configure OSDs on PVCs.
2020-06-05 08:54:45.905304 I | op-osd: start provisioning the osds on nodes, if needed
2020-06-05 08:54:45.956569 I | op-osd: 3 of the 3 storage nodes are valid
2020-06-05 08:54:46.049152 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-kube0 to start a new one
2020-06-05 08:54:46.126923 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube0 still exists
2020-06-05 08:54:48.152639 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube0 deleted
2020-06-05 08:54:48.189604 I | op-osd: osd provision job started for node kube0
2020-06-05 08:54:48.347086 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-kube1 to start a new one
2020-06-05 08:54:48.440373 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube1 still exists
2020-06-05 08:54:50.578751 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube1 deleted
2020-06-05 08:54:50.699504 I | op-osd: osd provision job started for node kube1
2020-06-05 08:54:50.851792 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-kube2 to start a new one
2020-06-05 08:54:50.965990 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube2 still exists
2020-06-05 08:54:53.152582 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube2 deleted
2020-06-05 08:54:53.222155 I | op-osd: osd provision job started for node kube2
2020-06-05 08:54:53.222579 I | op-osd: start osds after provisioning is completed, if needed
2020-06-05 08:54:53.264603 I | op-osd: osd orchestration status for node kube0 is starting
2020-06-05 08:54:53.265438 I | op-osd: osd orchestration status for node kube1 is starting
2020-06-05 08:54:53.267866 I | op-osd: osd orchestration status for node kube2 is starting
2020-06-05 08:54:53.268341 I | op-osd: 0/3 node(s) completed osd provisioning, resource version 74016482
2020-06-05 08:54:57.029214 I | op-mgr: successful modules: crash
2020-06-05 08:54:57.761554 I | op-osd: osd orchestration status for node kube0 is computingDiff
2020-06-05 08:54:58.277032 I | op-osd: osd orchestration status for node kube0 is orchestrating
2020-06-05 08:54:59.620328 I | op-osd: osd orchestration status for node kube1 is computingDiff
2020-06-05 08:55:00.116173 I | op-osd: osd orchestration status for node kube1 is orchestrating
2020-06-05 08:55:01.383081 I | op-osd: osd orchestration status for node kube0 is completed
2020-06-05 08:55:01.383169 I | op-osd: starting 1 osd daemons on node kube0
2020-06-05 08:55:01.383225 I | cephclient: getting or creating ceph auth key "osd.2"
2020-06-05 08:55:01.863243 I | op-mgr: successful modules: prometheus
2020-06-05 08:55:04.534943 I | op-mgr: the dashboard secret was already generated
2020-06-05 08:55:04.536925 I | op-mgr: setting ceph dashboard "admin" login creds
2020-06-05 08:55:23.372584 I | op-osd: deployment for osd 2 already exists. updating if needed
2020-06-05 08:55:23.748566 I | op-k8sutil: deployment "rook-ceph-osd-2" did not change, nothing to update
2020-06-05 08:55:23.749149 I | op-osd: started deployment for osd 2
2020-06-05 08:55:23.903466 I | op-osd: osd orchestration status for node kube1 is completed
2020-06-05 08:55:23.905251 I | op-osd: starting 0 osd daemons on node kube1
2020-06-05 08:55:24.126799 I | op-osd: osd orchestration status for node kube2 is computingDiff
2020-06-05 08:55:26.027287 I | op-osd: osd orchestration status for node kube2 is orchestrating
2020-06-05 08:55:33.281708 I | op-mgr: successfully set ceph dashboard creds
2020-06-05 08:55:36.536957 I | op-mgr: successful modules: mgr module(s) from the spec
2020-06-05 08:55:36.996358 I | op-mgr: successful modules: orchestrator modules
2020-06-05 08:56:26.040459 I | op-osd: waiting on orchestration status update from 1 remaining nodes
2020-06-05 08:56:35.868091 I | op-osd: osd orchestration status for node kube2 is failed
2020-06-05 08:56:35.868296 E | op-osd: orchestration for node kube2 failed: &{OSDs:[] Status:failed PvcBackedOSD:false Message:failed to configure devices: failed to initialize devices: failed ceph-volume: exit status 1}
2020-06-05 08:56:35.868704 I | op-osd: 3/3 node(s) completed osd provisioning
2020-06-05 08:56:35.871433 E | op-cluster: failed to update cluster in namespace "rook-ceph". failed to start the osds: 1 failures encountered while running osds in namespace rook-ceph: orchestration for node kube2 failed: &{OSDs:[] Status:failed PvcBackedOSD:false Message:failed to configure devices: failed to initialize devices: failed ceph-volume: exit status 1}
2020-06-05 08:56:40.235289 I | op-mgr: successful modules: http bind settings
2020-06-05 08:56:43.203759 I | op-mgr: dashboard config has changed. restarting the dashboard module.
2020-06-05 08:56:43.203909 I | op-mgr: restarting the mgr module
2020-06-05 08:56:56.474911 I | op-mgr: successful modules: dashboard
```

The disks on kube2 after OSD prepare

```
root@kube2:~# blkid
/dev/sda1: UUID="0548da66-7c3a-4a4e-83a0-287d28ddeb2d" TYPE="xfs" PARTUUID="7a91217b-01"
/dev/rbd0: UUID="eeb5913f-bb67-4682-940c-cb547954f719" TYPE="xfs"
/dev/rbd1: UUID="2593e640-8aa5-46c4-b6f6-0e26c9d8812e" TYPE="xfs"
/dev/sda2: UUID="fwIfDQ-Lijs-QlWh-MeYb-do9A-BZbC-xyrdSQ" TYPE="LVM2_member" PARTUUID="7a91217b-02"
root@kube2:~# lsblk
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda      8:0    0   477G  0 disk 
├─sda1   8:1    0  93.1G  0 part /
└─sda2   8:2    0 383.8G  0 part 
  └─ceph--21a0485c--3fea--4c9a--b5de--ebb0858d944f-osd--data--7f2ad539--4901--4973--a502--4a5a5567202c
       253:0    0 383.8G  0 lvm  
rbd0   252:0    0    10G  0 disk /var/lib/kubelet/pods/cb64b441-8dc3-4ec7-bbf9-e5a69fcb307d/volumes/ceph.rook.io~
rbd1   252:16   0    25G  0 disk /var/lib/kubelet/pods/cb64b441-8dc3-4ec7-bbf9-e5a69fcb307d/volumes/ceph.rook.io~
```

The output of OSD parepre

```
2020-06-05 08:56:38.347747 I | rookcmd: starting Rook v1.3.5 with arguments '/rook/rook ceph osd provision'
2020-06-05 08:56:38.349039 I | rookcmd: flag values: --cluster-id=d10e9428-72d1-44bc-afc6-9316d71563be, --data-device-filter=all, --data-device-path-filter=, --data-devices=, --encrypted-device=false, --force-format=false, --help=false, --location=, --log-flush-frequency=5s, --log-level=DEBUG, --metadata-device=, --node-name=kube2, --operator-image=, --osd-database-size=1024, --osd-store=, --osd-wal-size=576, --osds-per-device=1, --pvc-backed-osd=false, --service-account=
2020-06-05 08:56:38.350030 I | op-mon: parsing mon endpoints: q=10.109.190.149:6789,r=10.107.121.159:6789,m=10.110.175.97:6789,n=10.109.159.46:6789
2020-06-05 08:56:38.423545 I | op-osd: CRUSH location=root=default host=kube2
2020-06-05 08:56:38.424088 I | cephcmd: crush location of osd: root=default host=kube2
2020-06-05 08:56:38.490062 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-06-05 08:56:38.491996 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2020-06-05 08:56:38.493632 D | cephosd: config file @ /etc/ceph/ceph.conf: [global]
fsid                = 3434ddcd-53b8-431c-b87a-236a8fccd33f
mon initial members = n q r m
mon host            = [v2:10.109.159.46:3300,v1:10.109.159.46:6789],[v2:10.109.190.149:3300,v1:10.109.190.149:6789],[v2:10.107.121.159:3300,v1:10.107.121.159:6789],[v2:10.110.175.97:3300,v1:10.110.175.97:6789]
public addr         = 192.168.2.100
cluster addr        = 192.168.2.100

[client.admin]
keyring = /var/lib/rook/rook-ceph/client.admin.keyring

2020-06-05 08:56:38.494412 I | cephosd: discovering hardware
2020-06-05 08:56:38.494796 D | exec: Running command: lsblk --all --noheadings --list --output KNAME
2020-06-05 08:56:38.523780 D | exec: Running command: lsblk /dev/loop1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.539374 W | inventory: skipping device "loop1" because 'lsblk' failed. exit status 1
2020-06-05 08:56:38.539529 W | inventory: skipping rbd device "rbd0"
2020-06-05 08:56:38.539599 D | exec: Running command: lsblk /dev/loop6 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.552437 W | inventory: skipping device "loop6" because 'lsblk' failed. exit status 1
2020-06-05 08:56:38.553477 D | exec: Running command: lsblk /dev/loop4 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.572130 W | inventory: skipping device "loop4" because 'lsblk' failed. exit status 1
2020-06-05 08:56:38.572470 D | exec: Running command: lsblk /dev/loop2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.588225 W | inventory: skipping device "loop2" because 'lsblk' failed. exit status 1
2020-06-05 08:56:38.589712 W | inventory: skipping rbd device "rbd1"
2020-06-05 08:56:38.590383 D | exec: Running command: lsblk /dev/loop0 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.606471 W | inventory: skipping device "loop0" because 'lsblk' failed. exit status 1
2020-06-05 08:56:38.607654 D | exec: Running command: lsblk /dev/loop7 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.623393 W | inventory: skipping device "loop7" because 'lsblk' failed. exit status 1
2020-06-05 08:56:38.623501 D | exec: Running command: lsblk /dev/sda --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.640768 D | exec: Running command: sgdisk --print /dev/sda
2020-06-05 08:56:38.661247 D | exec: Running command: udevadm info --query=property /dev/sda
2020-06-05 08:56:38.677004 D | exec: Running command: lsblk --noheadings --pairs /dev/sda
2020-06-05 08:56:38.704925 I | inventory: skipping device "sda" because it has child, considering the child instead.
2020-06-05 08:56:38.705482 D | exec: Running command: lsblk /dev/sda2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.755550 D | exec: Running command: udevadm info --query=property /dev/sda2
2020-06-05 08:56:38.773507 D | exec: Running command: lsblk /dev/dm-0 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.792920 D | exec: Running command: sgdisk --print /dev/dm-0
2020-06-05 08:56:38.824854 D | exec: Running command: udevadm info --query=property /dev/dm-0
2020-06-05 08:56:38.845945 D | exec: Running command: lsblk /dev/sda1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.873273 D | exec: Running command: udevadm info --query=property /dev/sda1
2020-06-05 08:56:38.889677 D | exec: Running command: lsblk /dev/loop5 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.913885 W | inventory: skipping device "loop5" because 'lsblk' failed. exit status 1
2020-06-05 08:56:38.914354 D | exec: Running command: lsblk /dev/loop3 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 08:56:38.929976 W | inventory: skipping device "loop3" because 'lsblk' failed. exit status 1
2020-06-05 08:56:38.930137 D | inventory: discovered disks are [0xc000121440 0xc0001217a0 0xc00013e120]
2020-06-05 08:56:38.930161 I | cephosd: creating and starting the osds
2020-06-05 08:56:38.930244 D | cephosd: desiredDevices are [{Name:all OSDsPerDevice:1 MetadataDevice: DatabaseSizeMB:0 DeviceClass: IsFilter:true IsDevicePathFilter:false}]
2020-06-05 08:56:38.930301 D | cephosd: context.Devices are [0xc000121440 0xc0001217a0 0xc00013e120]
2020-06-05 08:56:38.930333 I | cephosd: skipping device "sda2" because it contains a filesystem "LVM2_member"
2020-06-05 08:56:38.930356 I | cephosd: skipping 'dm' device "dm-0"
2020-06-05 08:56:38.930380 I | cephosd: skipping device "sda1" because it contains a filesystem "xfs"
2020-06-05 08:56:38.981637 I | cephosd: configuring osd devices: {"Entries":{}}
2020-06-05 08:56:38.981773 I | cephosd: no new devices to configure. returning devices already configured with ceph-volume.
2020-06-05 08:56:38.981837 D | exec: Running command: ceph-volume lvm list  --format json
2020-06-05 08:56:41.112798 I | cephosd: 0 ceph-volume lvm osd devices configured on this node
2020-06-05 08:56:41.112921 W | cephosd: skipping OSD configuration as no devices matched the storage settings for this node "kube2"
```

Ceph status 

```
[root@kube12 /]# ceph status
  cluster:
    id:     3434ddcd-53b8-431c-b87a-236a8fccd33f
    health: HEALTH_WARN
            Degraded data redundancy: 7109/21847 objects degraded (32.540%), 32 pgs degraded, 32 pgs undersized
            8 daemons have recently crashed
            OSD count 2 < osd_pool_default_size 3
 
  services:
    mon: 4 daemons, quorum m,n,q,r (age 14m)
    mgr: a(active, since 4m)
    mds: sharedfs:1 {0=sharedfs-b=up:active} 1 up:standby-replay
    osd: 2 osds: 2 up (since 8h), 2 in (since 16h)
    rgw: 1 daemon active (objectstore.a)
 
  data:
    pools:   10 pools, 152 pgs
    objects: 7.37k objects, 23 GiB
    usage:   51 GiB used, 694 GiB / 745 GiB avail
    pgs:     7109/21847 objects degraded (32.540%)
             120 active+clean
             32  active+undersized+degraded
 
  io:
    client:   1.2 KiB/s rd, 102 KiB/s wr, 2 op/s rd, 5 op/s wr
```

Rook version

```
$ kubectl -n rook-ceph get deployments.apps rook-ceph-operator -o yaml | grep image:
        image: rook/ceph:v1.3.5
```

Note: Same behaviour with v1.3.4

kube2 is a master without taint:

```
$ kubectl describe node kube2
Name:               kube2
Roles:              master
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=kube2
                    kubernetes.io/os=linux
                    node-role.kubernetes.io/master=
Annotations:        kubeadm.alpha.kubernetes.io/cri-socket: /var/run/dockershim.sock
                    node.alpha.kubernetes.io/ttl: 0
                    projectcalico.org/IPv4Address: 10.10.2.22/24
                    projectcalico.org/IPv4IPIPTunnelAddr: 192.168.2.64
                    volumes.kubernetes.io/controller-managed-attach-detach: true
CreationTimestamp:  Thu, 04 Jun 2020 17:12:18 +0200
Taints:             <none>
Unschedulable:      false
Lease:
  HolderIdentity:  kube2
  AcquireTime:     <unset>
  RenewTime:       Fri, 05 Jun 2020 09:47:55 +0200
Conditions:
  Type                 Status  LastHeartbeatTime                 LastTransitionTime                Reason                       Message
  ----                 ------  -----------------                 ------------------                ------                       -------
  NetworkUnavailable   False   Fri, 05 Jun 2020 01:46:26 +0200   Fri, 05 Jun 2020 01:46:26 +0200   CalicoIsUp                   Calico is running on this node
  MemoryPressure       False   Fri, 05 Jun 2020 09:47:36 +0200   Fri, 05 Jun 2020 01:45:27 +0200   KubeletHasSufficientMemory   kubelet has sufficient memory available
  DiskPressure         False   Fri, 05 Jun 2020 09:47:36 +0200   Fri, 05 Jun 2020 01:45:27 +0200   KubeletHasNoDiskPressure     kubelet has no disk pressure
  PIDPressure          False   Fri, 05 Jun 2020 09:47:36 +0200   Fri, 05 Jun 2020 01:45:27 +0200   KubeletHasSufficientPID      kubelet has sufficient PID available
  Ready                True    Fri, 05 Jun 2020 09:47:36 +0200   Fri, 05 Jun 2020 01:45:28 +0200   KubeletReady                 kubelet is posting ready status. AppArmor enabled
Addresses:
  InternalIP:  10.10.2.22
  Hostname:    kube2
Capacity:
  cpu:                4
  ephemeral-storage:  97607104Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             4022756Ki
  pods:               110
Allocatable:
  cpu:                4
  ephemeral-storage:  89954706898
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             3920356Ki
  pods:               110
System Info:
  Machine ID:                 2e326a069ba540f7a2147621c336ea63
  System UUID:                2e326a069ba540f7a2147621c336ea63
  Boot ID:                    218ecc3f-8d06-46e0-9a09-898566ef32de
  Kernel Version:             4.15.0-101-generic
  OS Image:                   Ubuntu 18.04.4 LTS
  Operating System:           linux
  Architecture:               amd64
  Container Runtime Version:  docker://19.3.6
  Kubelet Version:            v1.16.10
  Kube-Proxy Version:         v1.16.10
PodCIDR:                      192.168.2.0/24
PodCIDRs:                     192.168.2.0/24
Non-terminated Pods:          (12 in total)
  Namespace                   Name                                  CPU Requests  CPU Limits  Memory Requests  Memory Limits  AGE
  ---------                   ----                                  ------------  ----------  ---------------  -------------  ---
  kube-system                 calico-node-6q86z                     250m (6%)     0 (0%)      0 (0%)           0 (0%)         16h
  kube-system                 coredns-5644d7b6d9-f9jcd              100m (2%)     0 (0%)      70Mi (1%)        170Mi (4%)     16h
  kube-system                 etcd-kube2                            0 (0%)        0 (0%)      0 (0%)           0 (0%)         16h
  kube-system                 kube-apiserver-kube2                  250m (6%)     0 (0%)      0 (0%)           0 (0%)         16h
  kube-system                 kube-controller-manager-kube2         200m (5%)     0 (0%)      0 (0%)           0 (0%)         16h
  kube-system                 kube-proxy-7zcns                      0 (0%)        0 (0%)      0 (0%)           0 (0%)         16h
  kube-system                 kube-scheduler-kube2                  100m (2%)     0 (0%)      0 (0%)           0 (0%)         16h
  monitoring                  node-exporter-lmt7c                   112m (2%)     270m (6%)   200Mi (5%)       220Mi (5%)     16h
  observium                   observium-655658dc8-hn4pd             0 (0%)        0 (0%)      0 (0%)           0 (0%)         15h
  rook-ceph                   rook-ceph-agent-hnds5                 0 (0%)        0 (0%)      0 (0%)           0 (0%)         8h
  rook-ceph                   rook-ceph-operator-d7797b858-w2x8j    0 (0%)        0 (0%)      0 (0%)           0 (0%)         7h35m
  rook-ceph                   rook-discover-krqmf                   0 (0%)        0 (0%)      0 (0%)           0 (0%)         8h
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests     Limits
  --------           --------     ------
  cpu                1012m (25%)  270m (6%)
  memory             270Mi (7%)   390Mi (10%)
  ephemeral-storage  0 (0%)       0 (0%)
  hugepages-1Gi      0 (0%)       0 (0%)
  hugepages-2Mi      0 (0%)       0 (0%)
Events:              <none>
```

**Comment by leseb** [comment]
I'm not sure what happens but a couple fo things are off:

1. This is not supported anymore:

```     
directories:
      - path: "/media/local"
```

So this won't do anything.

2. Somehow your disk was prepared by not the prepare pod log you shared.

3. not sure about this log since your prepare log does not show any errors:
```
2020-06-05 08:56:35.871433 E | op-cluster: failed to update cluster in namespace "rook-ceph". failed to start the osds: 1 failures encountered while running osds in namespace rook-ceph: orchestration for node kube2 failed: &{OSDs:[] Status:failed PvcBackedOSD:false Message:failed to configure devices: failed to initialize devices: failed ceph-volume: exit status 1}
```

Are you sure you pasted the correct prepare pod logs?

**Comment by dkrizic** [comment]
I am currently migrating from

```
directories:
      - path: "/media/local"
```

to using partitions. Once kube2 is migrated kube1 will be next and last. It worked on node kube0, but when I tried to migrate node kube2 I got stuck. This is the reason why I reported this issue. 

You are right, the osd-prepare did not create the LVM, this happened on a previous run. I will try again and report.

**Comment by dkrizic** [comment]
So, I cleaned the machine fully and run all again:

```
root@kube2:~# ls -lF /var/lib/rook
ls: cannot access '/var/lib/rook': No such file or directory
root@kube2:~# vgs
root@kube2:~# lvs
root@kube2:~# pvs
root@kube2:~# blkid
/dev/sda1: UUID="0548da66-7c3a-4a4e-83a0-287d28ddeb2d" TYPE="xfs" PARTUUID="7a91217b-01"
/dev/rbd0: UUID="eeb5913f-bb67-4682-940c-cb547954f719" TYPE="xfs"
/dev/rbd1: UUID="2593e640-8aa5-46c4-b6f6-0e26c9d8812e" TYPE="xfs"
/dev/sda2: PARTUUID="7a91217b-02"
root@kube2:~# lsblk
NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda      8:0    0   477G  0 disk 
├─sda1   8:1    0  93.1G  0 part /
└─sda2   8:2    0 383.8G  0 part 
rbd0   252:0    0    10G  0 disk /var/lib/kubelet/pods/cb64b441-8dc3-4ec7-bbf9-e5a69fcb307d/volumes/ceph.rook.io~r
rbd1   252:16   0    25G  0 disk /var/lib/kubelet/pods/cb64b441-8dc3-4ec7-bbf9-e5a69fcb307d/volumes/ceph.rook.io~r
```

Operator

```
2020-06-05 15:14:56.639506 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-kube0 to start a new one
2020-06-05 15:14:56.692340 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube0 still exists
2020-06-05 15:14:58.703103 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube0 deleted
2020-06-05 15:14:58.725457 I | op-osd: osd provision job started for node kube0
2020-06-05 15:14:58.848633 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-kube1 to start a new one
2020-06-05 15:14:58.922948 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube1 still exists
2020-06-05 15:15:00.287269 I | op-mgr: successful modules: prometheus
2020-06-05 15:15:00.305761 I | op-mgr: successful modules: crash
2020-06-05 15:15:00.975201 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube1 deleted
2020-06-05 15:15:01.142696 I | op-osd: osd provision job started for node kube1
2020-06-05 15:15:01.343639 I | op-k8sutil: Removing previous job rook-ceph-osd-prepare-kube2 to start a new one
2020-06-05 15:15:01.823026 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube2 still exists
2020-06-05 15:15:03.844657 I | op-k8sutil: batch job rook-ceph-osd-prepare-kube2 deleted
2020-06-05 15:15:03.891758 I | op-osd: osd provision job started for node kube2
2020-06-05 15:15:03.891808 I | op-osd: start osds after provisioning is completed, if needed
2020-06-05 15:15:03.952124 I | op-osd: osd orchestration status for node kube0 is starting
2020-06-05 15:15:03.952840 I | op-osd: osd orchestration status for node kube1 is starting
2020-06-05 15:15:03.962985 I | op-osd: osd orchestration status for node kube2 is starting
2020-06-05 15:15:03.964253 I | op-osd: 0/3 node(s) completed osd provisioning, resource version 74129184
2020-06-05 15:15:04.771120 I | op-mgr: successful modules: mgr module(s) from the spec
2020-06-05 15:15:05.088614 I | op-mgr: successful modules: orchestrator modules
2020-06-05 15:15:05.342486 I | op-mgr: the dashboard secret was already generated
2020-06-05 15:15:05.342541 I | op-mgr: setting ceph dashboard "admin" login creds
2020-06-05 15:15:07.889407 I | op-mgr: successfully set ceph dashboard creds
2020-06-05 15:15:08.357681 I | op-osd: osd orchestration status for node kube0 is computingDiff
2020-06-05 15:15:09.053752 I | op-osd: osd orchestration status for node kube0 is orchestrating
2020-06-05 15:15:09.917408 I | op-osd: osd orchestration status for node kube1 is computingDiff
2020-06-05 15:15:10.434900 I | op-osd: osd orchestration status for node kube1 is orchestrating
2020-06-05 15:15:12.839914 I | op-osd: osd orchestration status for node kube2 is computingDiff
2020-06-05 15:15:13.132132 I | op-osd: osd orchestration status for node kube1 is completed
2020-06-05 15:15:13.132190 I | op-osd: starting 0 osd daemons on node kube1
2020-06-05 15:15:13.309081 I | op-osd: osd orchestration status for node kube0 is completed
2020-06-05 15:15:13.309133 I | op-osd: starting 1 osd daemons on node kube0
2020-06-05 15:15:13.309189 I | cephclient: getting or creating ceph auth key "osd.2"
2020-06-05 15:15:15.244783 I | op-mgr: successful modules: http bind settings
2020-06-05 15:15:15.778289 I | op-osd: deployment for osd 2 already exists. updating if needed
2020-06-05 15:15:15.943358 I | op-k8sutil: deployment "rook-ceph-osd-2" did not change, nothing to update
2020-06-05 15:15:15.943425 I | op-osd: started deployment for osd 2
2020-06-05 15:15:20.331040 I | op-mgr: dashboard config has changed. restarting the dashboard module.
2020-06-05 15:15:20.331090 I | op-mgr: restarting the mgr module
2020-06-05 15:15:22.559622 I | op-osd: osd orchestration status for node kube2 is orchestrating
2020-06-05 15:15:24.950025 I | op-mgr: successful modules: dashboard
2020-06-05 15:16:05.818244 I | op-osd: osd orchestration status for node kube2 is failed
2020-06-05 15:16:05.818969 E | op-osd: orchestration for node kube2 failed: &{OSDs:[] Status:failed PvcBackedOSD:false Message:failed to configure devices: failed to initialize devices: failed ceph-volume: exit status 1}
2020-06-05 15:16:05.819074 I | op-osd: 3/3 node(s) completed osd provisioning
2020-06-05 15:16:05.819710 E | op-cluster: failed to create cluster in namespace "rook-ceph". failed to start the osds: 1 failures encountered while running osds in namespace rook-ceph: orchestration for node kube2 failed: &{OSDs:[] Status:failed PvcBackedOSD:false Message:failed to configure devices: failed to initialize devices: failed ceph-volume: exit status 1}
```

OSD Prepare

```
2020-06-05 15:16:07.764786 I | rookcmd: starting Rook v1.3.5 with arguments '/rook/rook ceph osd provision'
2020-06-05 15:16:07.765306 I | rookcmd: flag values: --cluster-id=d10e9428-72d1-44bc-afc6-9316d71563be, --data-device-filter=all, --data-device-path-filter=, --data-devices=, --encrypted-device=false, --force-format=false, --help=false, --location=, --log-flush-frequency=5s, --log-level=DEBUG, --metadata-device=, --node-name=kube2, --operator-image=, --osd-database-size=1024, --osd-store=, --osd-wal-size=576, --osds-per-device=1, --pvc-backed-osd=false, --service-account=
2020-06-05 15:16:07.765336 I | op-mon: parsing mon endpoints: n=10.109.159.46:6789,q=10.109.190.149:6789,r=10.107.121.159:6789,m=10.110.175.97:6789
2020-06-05 15:16:07.833183 I | op-osd: CRUSH location=root=default host=kube2
2020-06-05 15:16:07.833333 I | cephcmd: crush location of osd: root=default host=kube2
2020-06-05 15:16:07.890216 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-06-05 15:16:07.891159 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2020-06-05 15:16:07.892010 D | cephosd: config file @ /etc/ceph/ceph.conf: [global]
fsid                = 3434ddcd-53b8-431c-b87a-236a8fccd33f
mon initial members = m n q r
mon host            = [v2:10.110.175.97:3300,v1:10.110.175.97:6789],[v2:10.109.159.46:3300,v1:10.109.159.46:6789],[v2:10.109.190.149:3300,v1:10.109.190.149:6789],[v2:10.107.121.159:3300,v1:10.107.121.159:6789]
public addr         = 192.168.2.119
cluster addr        = 192.168.2.119

[client.admin]
keyring = /var/lib/rook/rook-ceph/client.admin.keyring

2020-06-05 15:16:07.892112 I | cephosd: discovering hardware
2020-06-05 15:16:07.892152 D | exec: Running command: lsblk --all --noheadings --list --output KNAME
2020-06-05 15:16:07.915554 D | exec: Running command: lsblk /dev/loop1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:07.931740 W | inventory: skipping device "loop1" because 'lsblk' failed. exit status 1
2020-06-05 15:16:07.932065 W | inventory: skipping rbd device "rbd0"
2020-06-05 15:16:07.932320 D | exec: Running command: lsblk /dev/loop6 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:07.948041 W | inventory: skipping device "loop6" because 'lsblk' failed. exit status 1
2020-06-05 15:16:07.948290 D | exec: Running command: lsblk /dev/loop4 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:07.962082 W | inventory: skipping device "loop4" because 'lsblk' failed. exit status 1
2020-06-05 15:16:07.962195 D | exec: Running command: lsblk /dev/loop2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:07.975148 W | inventory: skipping device "loop2" because 'lsblk' failed. exit status 1
2020-06-05 15:16:07.975350 W | inventory: skipping rbd device "rbd1"
2020-06-05 15:16:07.975535 D | exec: Running command: lsblk /dev/loop0 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:07.988299 W | inventory: skipping device "loop0" because 'lsblk' failed. exit status 1
2020-06-05 15:16:07.988486 D | exec: Running command: lsblk /dev/loop7 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:08.006133 W | inventory: skipping device "loop7" because 'lsblk' failed. exit status 1
2020-06-05 15:16:08.006258 D | exec: Running command: lsblk /dev/sda --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:08.022954 D | exec: Running command: sgdisk --print /dev/sda
2020-06-05 15:16:08.042205 D | exec: Running command: udevadm info --query=property /dev/sda
2020-06-05 15:16:08.064016 D | exec: Running command: lsblk --noheadings --pairs /dev/sda
2020-06-05 15:16:08.105177 I | inventory: skipping device "sda" because it has child, considering the child instead.
2020-06-05 15:16:08.105610 D | exec: Running command: lsblk /dev/sda2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:08.148415 D | exec: Running command: udevadm info --query=property /dev/sda2
2020-06-05 15:16:08.195828 D | exec: Running command: lsblk /dev/dm-0 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:08.225111 D | exec: Running command: sgdisk --print /dev/dm-0
2020-06-05 15:16:08.264132 D | exec: Running command: udevadm info --query=property /dev/dm-0
2020-06-05 15:16:08.290100 D | exec: Running command: lsblk /dev/sda1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:08.328301 D | exec: Running command: udevadm info --query=property /dev/sda1
2020-06-05 15:16:08.343943 D | exec: Running command: lsblk /dev/loop5 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:08.356684 W | inventory: skipping device "loop5" because 'lsblk' failed. exit status 1
2020-06-05 15:16:08.356852 D | exec: Running command: lsblk /dev/loop3 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:16:08.368683 W | inventory: skipping device "loop3" because 'lsblk' failed. exit status 1
2020-06-05 15:16:08.368930 D | inventory: discovered disks are [0xc00020f8c0 0xc000184000 0xc000184240]
2020-06-05 15:16:08.369013 I | cephosd: creating and starting the osds
2020-06-05 15:16:08.369097 D | cephosd: desiredDevices are [{Name:all OSDsPerDevice:1 MetadataDevice: DatabaseSizeMB:0 DeviceClass: IsFilter:true IsDevicePathFilter:false}]
2020-06-05 15:16:08.369153 D | cephosd: context.Devices are [0xc00020f8c0 0xc000184000 0xc000184240]
2020-06-05 15:16:08.369184 I | cephosd: skipping device "sda2" because it contains a filesystem "LVM2_member"
2020-06-05 15:16:08.369207 I | cephosd: skipping 'dm' device "dm-0"
2020-06-05 15:16:08.369352 I | cephosd: skipping device "sda1" because it contains a filesystem "xfs"
2020-06-05 15:16:08.407846 I | cephosd: configuring osd devices: {"Entries":{}}
2020-06-05 15:16:08.407952 I | cephosd: no new devices to configure. returning devices already configured with ceph-volume.
2020-06-05 15:16:08.407991 D | exec: Running command: ceph-volume lvm list  --format json
2020-06-05 15:16:09.861858 I | cephosd: 0 ceph-volume lvm osd devices configured on this node
2020-06-05 15:16:09.862649 W | cephosd: skipping OSD configuration as no devices matched the storage settings for this node "kube2"
```

State on the machine after this OSD prepare run

```
root@kube2:~# ls -lF /var/lib/rook/
total 0
drwxr-xr-x 4 root root 82 Jun  5 17:15 rook-ceph/NAME   MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
sda      8:0    0   477G  0 disk 
├─sda1   8:1    0  93.1G  0 part /
└─sda2   8:2    0 383.8G  0 part 
  └─ceph--c104326f--a53e--4245--9221--89cb2ba21533-osd--data--fc55e916--48ca--4ead--a4e8--1433dd614518
       253:0    0 383.8G  0 lvm  
rbd0   252:0    0    10G  0 disk /var/lib/kubelet/pods/cb64b441-8dc3-4ec7-bbf9-e5a69fcb307d/volumes/ceph.rook.io~r
rbd1   252:16   0    25G  0 disk /var/lib/kubelet/pods/cb64b441-8dc3-4ec7-bbf9-e5a69fcb307d/volumes/ceph.rook.io~r
root@kube2:~# blkid
/dev/sda1: UUID="0548da66-7c3a-4a4e-83a0-287d28ddeb2d" TYPE="xfs" PARTUUID="7a91217b-01"
/dev/rbd0: UUID="eeb5913f-bb67-4682-940c-cb547954f719" TYPE="xfs"
/dev/rbd1: UUID="2593e640-8aa5-46c4-b6f6-0e26c9d8812e" TYPE="xfs"
/dev/sda2: UUID="7RszFW-m3yL-Hy0A-78N4-dY3s-soXq-OVgwQZ" TYPE="LVM2_member" PARTUUID="7a91217b-02"
root@kube2:~# dmsetup ls
ceph--c104326f--a53e--4245--9221--89cb2ba21533-osd--data--fc55e916--48ca--4ead--a4e8--1433dd614518	(253:0)
root@kube2:~# vgs
root@kube2:~# lvs
root@kube2:~# pvs
```

What is strange that I need to restart the LVM daemons to see the actual LVM:

```
root@kube2:~# systemctl restart lvm2-lvmetad
root@kube2:~# systemctl restart lvm2-monitor
root@kube2:~# systemctl restart lvm2-lvmpolld
root@kube2:~# vgs
  VG                                        #PV #LV #SN Attr   VSize   VFree
  ceph-c104326f-a53e-4245-9221-89cb2ba21533   1   1   0 wz--n- 383.80g    0 
root@kube2:~# lvs
  LV                                            VG                                        Attr       LSize   Pool Origin Data%  Meta%  Move Log Cpy%Sync Convert
  osd-data-fc55e916-48ca-4ead-a4e8-1433dd614518 ceph-c104326f-a53e-4245-9221-89cb2ba21533 -wi-a----- 383.80g                                                    
root@kube2:~# pvs
  PV         VG                                        Fmt  Attr PSize   PFree
  /dev/sda2  ceph-c104326f-a53e-4245-9221-89cb2ba21533 lvm2 a--  383.80g    0 
root@kube2:~# dmsetup ls
ceph--c104326f--a53e--4245--9221--89cb2ba21533-osd--data--fc55e916--48ca--4ead--a4e8--1433dd614518	(253:0)
```

So either something on the machine constantly re-creates the LVM stuff or I do no remove the LVM stuff properly. I am no expert on this. Anybody who can give me some hints what to look for.

Is there a way to tell Rook-Ceph to use the partition /dev/sda2 as-is without LV,PV,VG?

**Comment by dkrizic** [comment]
Ok, I found the root cause of it

```
 Error EEXIST: entity osd.1 exists but key does not match
```

From this OSD prepare run where the LVM was created

```
2020-06-05 15:44:24.323072 I | rookcmd: starting Rook v1.3.5 with arguments '/rook/rook ceph osd provision'
2020-06-05 15:44:24.329224 I | rookcmd: flag values: --cluster-id=d10e9428-72d1-44bc-afc6-9316d71563be, --data-device-filter=all, --data-device-path-filter=, --data-devices=, --encrypted-device=false, --force-format=false, --help=false, --location=, --log-flush-frequency=5s, --log-level=DEBUG, --metadata-device=, --node-name=kube2, --operator-image=, --osd-database-size=1024, --osd-store=, --osd-wal-size=576, --osds-per-device=1, --pvc-backed-osd=false, --service-account=
2020-06-05 15:44:24.330245 I | op-mon: parsing mon endpoints: n=10.109.159.46:6789,q=10.109.190.149:6789,r=10.107.121.159:6789,m=10.110.175.97:6789
2020-06-05 15:44:24.496895 I | op-osd: CRUSH location=root=default host=kube2
2020-06-05 15:44:24.506522 I | cephcmd: crush location of osd: root=default host=kube2
2020-06-05 15:44:24.625568 I | cephconfig: writing config file /var/lib/rook/rook-ceph/rook-ceph.config
2020-06-05 15:44:24.626361 I | cephconfig: generated admin config in /var/lib/rook/rook-ceph
2020-06-05 15:44:24.627060 D | cephosd: config file @ /etc/ceph/ceph.conf: [global]
fsid                = 3434ddcd-53b8-431c-b87a-236a8fccd33f
mon initial members = n q r m
mon host            = [v2:10.109.159.46:3300,v1:10.109.159.46:6789],[v2:10.109.190.149:3300,v1:10.109.190.149:6789],[v2:10.107.121.159:3300,v1:10.107.121.159:6789],[v2:10.110.175.97:3300,v1:10.110.175.97:6789]
public addr         = 192.168.2.124
cluster addr        = 192.168.2.124

[client.admin]
keyring = /var/lib/rook/rook-ceph/client.admin.keyring

2020-06-05 15:44:24.627088 I | cephosd: discovering hardware
2020-06-05 15:44:24.627118 D | exec: Running command: lsblk --all --noheadings --list --output KNAME
2020-06-05 15:44:24.724697 D | exec: Running command: lsblk /dev/loop1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:24.777541 W | inventory: skipping device "loop1" because 'lsblk' failed. exit status 1
2020-06-05 15:44:24.777719 D | exec: Running command: lsblk /dev/loop6 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:24.814882 W | inventory: skipping device "loop6" because 'lsblk' failed. exit status 1
2020-06-05 15:44:24.815050 D | exec: Running command: lsblk /dev/loop4 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:24.852469 W | inventory: skipping device "loop4" because 'lsblk' failed. exit status 1
2020-06-05 15:44:24.854497 D | exec: Running command: lsblk /dev/loop2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:24.882624 W | inventory: skipping device "loop2" because 'lsblk' failed. exit status 1
2020-06-05 15:44:24.882798 D | exec: Running command: lsblk /dev/loop0 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:24.917507 W | inventory: skipping device "loop0" because 'lsblk' failed. exit status 1
2020-06-05 15:44:24.917652 D | exec: Running command: lsblk /dev/loop7 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:24.970110 W | inventory: skipping device "loop7" because 'lsblk' failed. exit status 1
2020-06-05 15:44:24.970273 D | exec: Running command: lsblk /dev/sda --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:25.059444 D | exec: Running command: sgdisk --print /dev/sda
2020-06-05 15:44:25.098885 D | exec: Running command: udevadm info --query=property /dev/sda
2020-06-05 15:44:25.140128 D | exec: Running command: lsblk --noheadings --pairs /dev/sda
2020-06-05 15:44:25.466251 I | inventory: skipping device "sda" because it has child, considering the child instead.
2020-06-05 15:44:25.466433 D | exec: Running command: lsblk /dev/sda2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:25.509652 D | exec: Running command: udevadm info --query=property /dev/sda2
2020-06-05 15:44:25.553857 D | exec: Running command: lsblk /dev/sda1 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:25.590016 D | exec: Running command: udevadm info --query=property /dev/sda1
2020-06-05 15:44:25.657796 D | exec: Running command: lsblk /dev/loop5 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:25.680603 W | inventory: skipping device "loop5" because 'lsblk' failed. exit status 1
2020-06-05 15:44:25.682086 D | exec: Running command: lsblk /dev/loop3 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:25.711304 W | inventory: skipping device "loop3" because 'lsblk' failed. exit status 1
2020-06-05 15:44:25.711457 D | inventory: discovered disks are [0xc00016efc0 0xc00016f440]
2020-06-05 15:44:25.711509 I | cephosd: creating and starting the osds
2020-06-05 15:44:25.711742 D | cephosd: desiredDevices are [{Name:all OSDsPerDevice:1 MetadataDevice: DatabaseSizeMB:0 DeviceClass: IsFilter:true IsDevicePathFilter:false}]
2020-06-05 15:44:25.711801 D | cephosd: context.Devices are [0xc00016efc0 0xc00016f440]
2020-06-05 15:44:25.711873 D | exec: Running command: lsblk /dev/sda2 --bytes --nodeps --pairs --paths --output SIZE,ROTA,RO,TYPE,PKNAME,NAME
2020-06-05 15:44:25.747784 D | exec: Running command: ceph-volume inventory --format json /dev/sda2
2020-06-05 15:44:36.819657 I | cephosd: device "sda2" is available.
2020-06-05 15:44:36.819810 I | cephosd: skipping device "sda1" because it contains a filesystem "xfs"
2020-06-05 15:44:36.958394 I | cephosd: configuring osd devices: {"Entries":{"sda2":{"Data":-1,"Metadata":null,"Config":{"Name":"","OSDsPerDevice":0,"MetadataDevice":"","DatabaseSizeMB":0,"DeviceClass":"","IsFilter":false,"IsDevicePathFilter":false},"PersistentDevicePaths":null}}}
2020-06-05 15:44:36.994822 I | cephclient: getting or creating ceph auth key "client.bootstrap-osd"
2020-06-05 15:44:36.996713 D | exec: Running command: ceph auth get-or-create-key client.bootstrap-osd mon allow profile bootstrap-osd --connect-timeout=15 --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring --format json --out-file /tmp/240599818
2020-06-05 15:44:44.671242 I | cephosd: configuring new device sda2
2020-06-05 15:44:44.671411 I | cephosd: Base command - stdbuf
2020-06-05 15:44:44.671463 I | cephosd: immediateReportArgs - stdbuf
2020-06-05 15:44:44.671540 I | cephosd: immediateExecuteArgs - [-oL ceph-volume lvm batch --prepare --bluestore --yes --osds-per-device 1 /dev/sda2]
2020-06-05 15:44:44.671572 D | exec: Running command: stdbuf -oL ceph-volume lvm batch --prepare --bluestore --yes --osds-per-device 1 /dev/sda2 --report
2020-06-05 15:44:52.259225 D | exec: 
2020-06-05 15:44:52.260804 D | exec: Total OSDs: 1
2020-06-05 15:44:52.261476 D | exec: 
2020-06-05 15:44:52.262138 D | exec:   Type            Path                                                    LV Size         % of device
2020-06-05 15:44:52.262769 D | exec: ----------------------------------------------------------------------------------------------------
2020-06-05 15:44:52.263813 D | exec:   [data]          /dev/sda2                                               382.00 GB       100%
2020-06-05 15:44:52.422438 D | exec: Running command: stdbuf -oL ceph-volume lvm batch --prepare --bluestore --yes --osds-per-device 1 /dev/sda2
2020-06-05 15:45:00.076236 D | exec: Running command: /usr/sbin/vgcreate --force --yes ceph-760333b8-124d-4b00-bdd4-eee3c16ae15f /dev/sda2
2020-06-05 15:45:00.663120 D | exec:  stderr: 
2020-06-05 15:45:00.664965 D | exec:  stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
2020-06-05 15:45:00.670210 D | exec:  stderr: 
2020-06-05 15:45:00.674636 D | exec:  stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
2020-06-05 15:45:00.689669 D | exec:  stderr: 
2020-06-05 15:45:00.692047 D | exec:  stderr: WARNING: Failed to connect to lvmetad. Falling back to device scanning.
2020-06-05 15:45:00.693564 D | exec:  stderr: 
2020-06-05 15:45:00.861358 D | exec:  stdout: Physical volume "/dev/sda2" successfully created.
2020-06-05 15:45:00.876578 D | exec:  stdout: Volume group "ceph-760333b8-124d-4b00-bdd4-eee3c16ae15f" successfully created
2020-06-05 15:45:01.671563 D | exec: Running command: /usr/sbin/lvcreate --yes -l 98254 -n osd-data-1cecce22-a7de-4b3e-959a-50153b283b2c ceph-760333b8-124d-4b00-bdd4-eee3c16ae15f
2020-06-05 15:45:02.267546 D | exec:  stderr: 
2020-06-05 15:45:02.270560 D | exec:  stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will manage logical volume symlinks in device directory.
2020-06-05 15:45:02.274932 D | exec:  stderr: 
2020-06-05 15:45:02.288449 D | exec:  stderr: 
2020-06-05 15:45:02.292737 D | exec:  stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, LVM will obtain device list by scanning device directory.
2020-06-05 15:45:02.315803 D | exec:  stderr: 
2020-06-05 15:45:02.317510 D | exec:  stderr: WARNING: Failed to connect to lvmetad. Falling back to device scanning.
2020-06-05 15:45:02.318678 D | exec:  stderr: 
2020-06-05 15:45:02.385257 D | exec:  stderr: 
2020-06-05 15:45:02.387170 D | exec:  stderr: Udev is running and DM_DISABLE_UDEV environment variable is set. Bypassing udev, device-mapper library will manage device nodes in device directory.
2020-06-05 15:45:02.389110 D | exec:  stderr: 
2020-06-05 15:45:02.454234 D | exec:  stdout: Logical volume "osd-data-1cecce22-a7de-4b3e-959a-50153b283b2c" created.
2020-06-05 15:45:10.767427 D | exec: Running command: /usr/bin/ceph-authtool --gen-print-key
2020-06-05 15:45:11.618235 D | exec: Running command: /usr/bin/ceph --cluster ceph --name client.bootstrap-osd --keyring /var/lib/ceph/bootstrap-osd/ceph.keyring -i - osd new 1b4a055e-816a-4e97-9a3c-a39550143580
2020-06-05 15:45:17.295577 D | exec:  stderr: Error EEXIST: entity osd.1 exists but key does not match
2020-06-05 15:45:17.369184 D | exec: Traceback (most recent call last):
2020-06-05 15:45:17.369269 D | exec:   File "/usr/sbin/ceph-volume", line 9, in <module>
2020-06-05 15:45:17.369292 D | exec:     load_entry_point('ceph-volume==1.0.0', 'console_scripts', 'ceph-volume')()
2020-06-05 15:45:17.369321 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/main.py", line 39, in __init__
2020-06-05 15:45:17.369342 D | exec:     self.main(self.argv)
2020-06-05 15:45:17.369363 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/decorators.py", line 59, in newfunc
2020-06-05 15:45:17.369388 D | exec:     return f(*a, **kw)
2020-06-05 15:45:17.369408 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/main.py", line 150, in main
2020-06-05 15:45:17.369648 D | exec:     terminal.dispatch(self.mapper, subcommand_args)
2020-06-05 15:45:17.369676 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/terminal.py", line 194, in dispatch
2020-06-05 15:45:17.369929 D | exec:     instance.main()
2020-06-05 15:45:17.370276 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/main.py", line 42, in main
2020-06-05 15:45:17.370911 D | exec:     terminal.dispatch(self.mapper, self.argv)
2020-06-05 15:45:17.371115 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/terminal.py", line 194, in dispatch
2020-06-05 15:45:17.371450 D | exec:     instance.main()
2020-06-05 15:45:17.371598 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/decorators.py", line 16, in is_root
2020-06-05 15:45:17.371888 D | exec:     return func(*a, **kw)
2020-06-05 15:45:17.372222 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/batch.py", line 325, in main
2020-06-05 15:45:17.372839 D | exec:     self.execute()
2020-06-05 15:45:17.373195 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/batch.py", line 288, in execute
2020-06-05 15:45:17.373897 D | exec:     self.strategy.execute()
2020-06-05 15:45:17.374305 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/strategies/bluestore.py", line 122, in execute
2020-06-05 15:45:17.374925 D | exec:     Prepare(command).main()
2020-06-05 15:45:17.375388 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/prepare.py", line 397, in main
2020-06-05 15:45:17.376089 D | exec:     self.safe_prepare()
2020-06-05 15:45:17.376477 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/prepare.py", line 246, in safe_prepare
2020-06-05 15:45:17.377270 D | exec:     self.prepare()
2020-06-05 15:45:17.377996 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/decorators.py", line 16, in is_root
2020-06-05 15:45:17.378693 D | exec:     return func(*a, **kw)
2020-06-05 15:45:17.379060 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/devices/lvm/prepare.py", line 286, in prepare
2020-06-05 15:45:17.379994 D | exec:     self.osd_id = prepare_utils.create_id(osd_fsid, json.dumps(secrets), osd_id=self.args.osd_id)
2020-06-05 15:45:17.380603 D | exec:   File "/usr/lib/python2.7/site-packages/ceph_volume/util/prepare.py", line 170, in create_id
2020-06-05 15:45:17.381425 D | exec:     raise RuntimeError('Unable to create a new OSD id')
2020-06-05 15:45:17.382672 D | exec: RuntimeError: Unable to create a new OSD id
failed to configure devices: failed to initialize devices: failed ceph-volume: exit status 1
```

This seems to be a core Ceph problem, not Rook related right? Feel free to close to confirm.

**Comment by dkrizic** [comment]
A

```
ceph auth rm osd.1
```

did the trick.
